pkg/logtailing/tailer.go

```diff
@@ -105,85 +108,134 @@ func New(cfg *Config) *Tailer {
 }
 
 // Run sets the websocket connection
-func (tailer *Tailer) Run() error {
-	s := ansi.StartSpinner("Getting ready...", tailer.cfg.Log.Out)
+func (t *Tailer) Run() error {
+	s := ansi.StartSpinner("Getting ready...", t.cfg.Log.Out)
 
-	// Intercept Ctrl+c so we can do some clean up
-	signal.Notify(tailer.interruptCh, os.Interrupt, syscall.SIGTERM)
+	// Create a context that will be canceled when Ctrl+C is pressed
+	ctx, cancel := context.WithCancel(context.Background())
+	t.ctx = ctx
+	signal.Notify(t.interruptCh, os.Interrupt, syscall.SIGTERM)
 
-	filters, err := jsonifyFilters(tailer.cfg.Filters)
-	if err != nil {
-		tailer.cfg.Log.Fatalf("Error while converting log filters to JSON encoding: %v", err)
-	}
+	go func() {
+		log.WithFields(log.Fields{
+			"prefix": "logtailing.Tailer.Run",
+		}).Debug("Ctrl+C received, cleaning up...")
 
-	session, err := tailer.stripeAuthClient.Authorize(tailer.cfg.DeviceName, tailer.cfg.WebSocketFeature, &filters)
+		<-t.interruptCh
+		cancel()
+	}()
+
+	// Create the CLI session
+	session, err := t.createSession()
 	if err != nil {
-		tailer.cfg.Log.Fatalf("Error while authenticating with Stripe: %v", err)
+		ansi.StopSpinner(s, "", t.cfg.Log.Out)
+		t.cfg.Log.Fatalf("Error while authenticating with Stripe: %v", err)
 	}
 
-	tailer.webSocketClient = websocket.NewClient(
+	// Create and start the websocket client
+	t.webSocketClient = websocket.NewClient(
 		session.WebSocketURL,
 		session.WebSocketID,
 		session.WebSocketAuthorizedFeature,
 		&websocket.Config{
-			EventHandler:      websocket.EventHandlerFunc(tailer.processRequestLogEvent),
-			Log:               tailer.cfg.Log,
-			NoWSS:             tailer.cfg.NoWSS,
+			EventHandler:      websocket.EventHandlerFunc(t.processRequestLogEvent),
+			Log:               t.cfg.Log,
+			NoWSS:             t.cfg.NoWSS,
 			ReconnectInterval: time.Duration(session.ReconnectDelay) * time.Second,
 		},
 	)
-	go tailer.webSocketClient.Run()
-
-	ansi.StopSpinner(s, "Ready! You're now waiting to receive API request logs (^C to quit)", tailer.cfg.Log.Out)
+	go t.webSocketClient.Run()
+
+	select {
+	case <-t.webSocketClient.Connected():
+		ansi.StopSpinner(s, "Ready! You're now waiting to receive API request logs (^C to quit)", t.cfg.Log.Out)
+	case <-t.ctx.Done():
+		ansi.StopSpinner(s, "", t.cfg.Log.Out)
+		t.cfg.Log.Fatalf("Aborting")
+	}
 
 	if session.DisplayConnectFilterWarning {
 		color := ansi.Color(os.Stdout)
-		fmt.Println(fmt.Sprintf("%s you specified the 'account' filter for connect accounts but are not a connect merchant, so the filter will not be applied.", color.Yellow("Warning")))
+		fmt.Println(fmt.Sprintf("%s you specified the 'account' filter for Connect accounts but are not a Connect merchant, so the filter will not be applied.", color.Yellow("Warning")))
```

Comment:
I know this is unrelated but could we change this to: "Warning: you specified the 'account' filter which applies only to Connect users so it will have no affect."

We don't really talk about `merchants` as a concept externally.